datavault-webapp/src/main/java/org/datavaultplatform/webapp/controllers/admin/AdminRolesController.java

```diff
@@ -0,0 +1,307 @@
+package org.datavaultplatform.webapp.controllers.admin;
+
+import org.datavaultplatform.common.model.PermissionModel;
+import org.datavaultplatform.common.model.RoleModel;
+import org.datavaultplatform.common.model.RoleType;
+import org.datavaultplatform.webapp.services.RestService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.datavaultplatform.webapp.model.RoleViewModel;
+
+import java.util.*;
+//import java.util.stream.Collectors;
+
+
+@Controller
+public class AdminRolesController {
+
+    private static final Logger logger = LoggerFactory.getLogger(AdminRolesController.class);
+    private RestService restService;
+
+    public void setRestService(RestService restService) {
+        this.restService = restService;
+    }
+
+    private void setCommonModel(ModelMap model) {
+
+
+        List<RoleModel> roles = restService.getEditableRoles();
+
+        if (roles.isEmpty()) {
+            logger.info("getRolesListing: roles has no items in rest fetch");
+        }
+
+        model.addAttribute("roles", roles);
+
+
+        if (!model.containsAttribute("errormessage")) {
+            model.addAttribute("errormessage", "");
+        }
+
+
+    }
+
+
+    @RequestMapping(value = "/admin/roles", method = RequestMethod.GET)
+    public String getRolesListing(ModelMap model) {
```

Comment:
There's a slightly more explicit way of doing these controllers, returning a `ModelAndView` object so I'm going to refactor these to use that instead